src/components/TimeRangePicker/TimeRangePickerBody.js

```diff
@@ -0,0 +1,204 @@
+import PropTypes from "prop-types";
+import clsx from "clsx";
+import { useEffect, useState } from "react";
+import dayjs from "dayjs";
+import { TimeRangeList } from "./TimeRangeList";
+import "./index.css";
+import { storage, createValueForInput, convertRangeValue } from "./helpers";
+import { RecentlyRanges } from "./RecentlyRanges";
+import { displayTimeFormat } from "./rangeOptions";
+import { TimePickerCalendar } from "./TimePickerCalendar";
+import { TimePickerInput } from "./TimePickerInput";
+
+export const TimeRangePickerBody = ({
+  isOpen,
+  closePicker,
+  currentRange,
+  changeRangeValue
+}) => {
+  const [recentRanges, setRecentRanges] = useState(
+    storage.getItem("timePickerRanges") || []
+  );
+  const [showCalendar, setShowCalendar] = useState(false);
+  const [calendarValue, setCalendarValue] = useState(null);
+  const [inputValueFrom, setInputValueFrom] = useState(
+    createValueForInput(currentRange.from)
+  );
+  const [inputValueTo, setInputValueTo] = useState(
+    createValueForInput(currentRange.to)
+  );
+  const [errorInputFrom, setErrorInputFrom] = useState(null);
+  const [errorInputTo, setErrorInputTo] = useState(null);
+
+  const changeRecentRangesList = (range) => {
+    if (
+      !recentRanges.find(
+        (el) =>
+          dayjs(el.from).toISOString() === dayjs(range.from).toISOString() &&
+          dayjs(el.to).toISOString() === dayjs(range.to).toISOString()
+      )
+    ) {
+      let newRanges;
+      if (recentRanges.length < 4) {
+        newRanges = [range, ...recentRanges];
+      } else {
+        newRanges = [range, ...recentRanges.slice(0, 3)];
+      }
+      setRecentRanges([...newRanges]);
+      storage.setItem("timePickerRanges", newRanges);
+    }
+  };
+
+  const onChangeCalendarRange = (value) => {
```

Comment:
using `useCallback`

Reply:
weird, i used it here and it shows up in modified files, its done